client/server/render/index.js

```diff
@@ -233,6 +233,7 @@ export function serverRender( req, res ) {
 		attachHead( context );
 
 		const cacheableReduxSubtrees = [ 'documentHead' ];
+		const cacheableLoggedOutReduxSubtrees = ! context.user ? [ 'themes', 'ui' ] : [];
```

Comment:
Nit: I'd just always define the array, and then only cache those elements if we're logged out. (It's not like the "cacheable logged out subtrees" ever change.)

Finally, I'm slightly worried about adding `ui` to the cache. I don't fully know the implications for or against, but from my testing, I think we should _only_ cache data which can be 1:1 matched to a route based on the cache key. For example, the themes data is the same on every visit to `/fr/themes`, and it's always the same on every visit to `/themes`. But I'm not sure if that's also the case for `ui`. 

In my testing, caching `ui` also didn't lead to a meaningful improvement in performance on the `themes` route.

The only reason I'm worried is based on my explanation in #66431. These were removed from the cache on purpose (my guess is to play nicer with locale, though the change wasn't explained). So I just want to be careful about regressions since they _used_ to be cached 5 years ago.